core/src/main/java/com/transferwise/common/gracefulshutdown/strategies/TaskSchedulersGracefulShutdownStrategy.java

```diff
@@ -58,6 +59,8 @@ public void prepareForShutdown() {
               scheduledThreadPoolExecutor.setExecuteExistingDelayedTasksAfterShutdownPolicy(false);
               scheduledThreadPoolExecutor.getQueue().clear();
               scheduledThreadPoolExecutor.shutdown();
+            } else if (executor instanceof ExecutorService) {
+              ((ExecutorService) executor).shutdown();
```

Comment:
since it's a shared library, I'd add a unit test to validate this use case (with `DelegatedExecutorService`)

Reply:
added unit test now 🙇